File: InsertionSort.java


public class InsertionSort {
    public static void main(String[] args) {
        int []a= {3,5,2,6,1,4};
        int j,x;
        for (int i=1; i<a.length; i++){
            j = i-1;
            x=a[i];
            while (j>-1 && a[j]>x){
                a[j+1]=a[j];
                j--;
            }
            a[j+1]=x;
        }
        for (int i = 0; i<a.length; i++){
            System.out.print(a[i]+" ");
        }
    }
}

File: insertion_sort.py
def insertionSort(arr):
 
    # Traverse through 1 to len(arr)
    for i in range(1, len(arr)):
 
        key = arr[i]
        j = i-1
        while j >=0 and key < arr[j] :
                arr[j+1] = arr[j]
                j -= 1
        arr[j+1] = key
 
 
arr = [22,99,79,999,20202]
insertionSort(arr)
#empty list to store sorted elements
lst = [] 
print("The Sorted array is : ")
for i in range(len(arr)):
    #appending the elements in sorted order
    lst.append(arr[i])  
print(lst)
